docs/datastores.md

```diff
@@ -0,0 +1,76 @@
+# Datastore Configuration Options
+
+This document describes the different possible values for the `Datastore.Spec`
+field in the ipfs configuration file.
+
+## flatfs
+Stores each key value pair as a file on the filesystem.
```

Comment:
Might be worth mentioning that flatfs should only be used for `/blocks` as it's `.Query` isn't complete.